lightning/src/ln/channel.rs

```diff
@@ -1144,7 +1137,7 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	///
 	/// Note that it is still possible to hit these assertions in case we find a preimage on-chain
 	/// but then have a reorg which settles on an HTLC-failure on chain.
-	fn get_update_fulfill_htlc(&mut self, htlc_id_arg: u64, payment_preimage_arg: PaymentPreimage) -> Result<(Option<msgs::UpdateFulfillHTLC>, Option<ChannelMonitorUpdate>), ChannelError> {
+	fn get_update_fulfill_htlc<L: Deref>(&mut self, htlc_id_arg: u64, payment_preimage_arg: PaymentPreimage, logger: &L) -> Result<(Option<msgs::UpdateFulfillHTLC>, Option<ChannelMonitorUpdate>), ChannelError> where L::Target: Logger {
```

Comment:
This (and a thousand other places) probably doesn't need to be a Deref - we pass in a reference to the thing, so might as well just dereference it at the callsite and pass in L: Logger.

Reply:
Hm, it looks like that causes this error: https://i.imgur.com/2wdqfaX.png Which seemingly would be an issue for e.g. the fuzz crate's TestLogger, since it contains a pointer to a trait object?

Reply:
Ah, alright. Funny that rust can't figure out how to do it without the "&*", but, totally fine to leave it as-is if it doesnt work.